examples/turnip-town/move/sources/game.move

```diff
@@ -0,0 +1,157 @@
+    /// already been deleted.
+    public fun burn(deed: Deed, game: &mut Game) {
+        let fid = field::deed_field(&deed);
+        assert!(table::contains(&game.fields, fid), ENoSuchField);
+
+        let field = table::remove(&mut game.fields, fid);
+        field::burn_field(field);
+        field::burn_deed(deed);
+    }
+
+    /// Sow a turnip at position (i, j) of the field owned by `deed`
+    /// in `game`.
+    ///
+    /// Fails if the field does not exist for this `deed`, or there is
+    /// already a turnip at that position.
+    public fun sow(
+        deed: &Deed,
+        game: &mut Game,
+        i: u64,
+        j: u64,
+        ctx: &mut TxContext,
+    ) {
+        let fid = field::deed_field(deed);
+        assert!(table::contains(&game.fields, fid), ENoSuchField);
+
+        let field = table::borrow_mut(&mut game.fields, fid);
+        field::sow(field, i, j, ctx);
+    }
+
+    /// Water the field owned by `deed`.
+    ///
+    /// Fails if the field does not exist for this `deed`.
+    public fun water(deed: &Deed, game: &mut Game) {
+        let fid = field::deed_field(deed);
+        assert!(table::contains(&game.fields, fid), ENoSuchField);
+
+        let field = table::borrow_mut(&mut game.fields, fid);
+        field::water(field, WATER_INCREMENT);
+    }
+
+    /// Harvest a turnip at position (i, j) of the field owned by
+    /// `deed`.
+    ///
+    /// Fails if the field for this `deed` does not exist, there is no
+    /// turnip to harvest at this position or that turnip is too small
+    /// to harvest.
+    public fun harvest(deed: &Deed, game: &mut Game, i: u64, j: u64): Turnip {
+        let fid = field::deed_field(deed);
+        assert!(table::contains(&game.fields, fid), ENoSuchField);
+
+        let field = table::borrow_mut(&mut game.fields, fid);
+        field::harvest(field, i, j)
+    }
+
+    /// Admin-only operation for the game service to simulate weather.
+    ///
+    /// Rain is simulated by adding `rain_amount` to the field's water
+    /// supply. The water supply is then shared among all turnips in
+    /// the field.
+    ///
+    /// Turnips gain 5% freshness (up to a max of 100%) for each
+    /// simulation tick they have enough water for.
+    ///
+    /// Every turnip needs water equal to its size to remain fresh.
+    /// If there is not enough water to keep all turnips fresh, then
+    /// freshness halves (turnips dry).
+    ///
+    /// If it `is_sunny` and there is water left over, it is
+    /// distributed evenly between the turnips, to grow them.  Every 2
+    /// units of water increases turnip size by 1, up to 10 size
+    /// units.
+    ///
+    /// If there is water left over, then freshness also halves
+    /// (turnips rot).
+    ///
+    /// If freshness drops to zero, the turnip has died and will be
+    /// removed.
+    ///
+    /// Fails if the field (represented by its ID in the game's table)
+    /// does not exist.
+    public fun simulate_weather(
```

Comment:
I went back and forth on whether this should be an `entry` function.  The main benefit of doing that is to not have to commit to it being part of the module interface, but in the end it's not feasible to do that, because this function is going to be called against multiple `fid`s in the same `Game` and he rule about `entry` functions not accepting a previously modified input is going to get in the way of that.

Overall it seems like too many things might be coupled under the notion of `entry` function here?